pyvista/core/pointset.py

```diff
@@ -406,28 +412,116 @@ def is_all_triangles(self):
         --------
         >>> import pyvista
         >>> plane = pyvista.Plane()
-        >>> plane.is_all_triangles()
+        >>> plane.is_all_triangles
         False
 
         >>> sphere = pyvista.Sphere()
-        >>> sphere.is_all_triangles()
+        >>> sphere.is_all_triangles
         True
 
         """
+        class CallableBool(int):  # pragma: no cover
```

Comment:
I have some purity-based misgivings with this type. Practicality probably beats my concerns... but still:
1. what if someone downstream does `if poly.is_all_triangles is True`? This will no longer be true. Heck, `isinstance(poly.is_all_triangles, bool)` will also no longer be true. Interestingly, `poly.is_all_triangles()` _will_ be a proper bool... So this smells more like keeping the callable as the "real" API, and allowing the CallableBool as a safety net for those who try to use it as a property.

2. these "bools" are also not singletons. But this probably isn't a practical problem, because CallableFalse always hashes to 0 thanks to its intness, so putting multiple CallableBools in a set will only give you one at the end.

3. these "bools" can't be instantiated without an argument, unlike `bool()` which is `False`. But again this is probably fine because this class is for internal use. Although technically one could do `type(poly.is_all_triangles)()` to try instantiating a new one.

Reply:
These are all good points.  Really, the right answer would be to have not used a property as something callable.  The question is how to change this without messing up workflows, and I'm not able to come up with an elegant solution since python isn't polymorphic (though you can cheat).

The best we can do at this point is to create a class that is nearly like a bool.  I'd change the API outright, but such a depreciation would instantly break any workflows that reply on this method and it wouldn't be clear why as they'd get:

```
TypeError: 'bool' object is not callable
```

I'll clean up this class the best I can, but for this I think:
```
Special cases aren't special enough to break the rules.
Although practicality beats purity.
```